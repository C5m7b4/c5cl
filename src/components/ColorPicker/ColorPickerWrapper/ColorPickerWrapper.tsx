import React, { useState, useEffect } from 'react';
import { Color } from '../../../types';
import themes from '../Theme/ColorTheme';
import Modal from '../Modal/Modal';
import ColorPicker from '../ColorPicker/ColorPicker';

export type ColorPickerWrapperProps = {
  color: Color;
  onChange: (color: Color) => void;
  height?: number;
  width?: number;
  borderRadius?: number;
  borderWidth?: number;
  borderStyle?: string;
  borderColor?: Color;
};

const ColorPickerWrapper: React.FC<ColorPickerWrapperProps> = ({
  color,
  onChange,
  height = 40,
  width = 120,
  borderRadius = 10,
  borderWidth = 5,
  borderStyle = 'solid',
  borderColor = '#333',
}) => {
  const [showPicker, setShowPicker] = useState(false);
  const [top, setTop] = useState(0);
  const [left, setLeft] = useState(0);

  useEffect(() => {}, []);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
    setTop(event.clientY - window.innerHeight / 2);
    setLeft(event.clientX - window.innerWidth / 2);
    setShowPicker(!showPicker);
  };

  const swatchStyle = {
    height: `${height}px`,
    width: `${width}px`,
    borderRadius: `${borderRadius}px`,
    borderWidth: `${borderWidth}px`,
    borderColor: `${borderColor}`,
    borderStyle: `${borderStyle}`,
    cursor: 'pointer',
    backgroundColor: `${color}`,
  };

  return (
    <div>
      <button
        className="mpw-swatch"
        style={swatchStyle}
        onClick={handleClick}
      ></button>
      {showPicker && (
        <Modal show={showPicker} onClose={() => setShowPicker(false)}>
          <ColorPicker
            color={color}
            // @ts-ignore
            onChange={(color) => onChange(color.hex)}
            theme={themes.dark}
            top={top}
            left={left}
          />
        </Modal>
      )}
    </div>
  );
};

export default ColorPickerWrapper;
