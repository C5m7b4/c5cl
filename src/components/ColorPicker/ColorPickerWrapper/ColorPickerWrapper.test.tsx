import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ColorPickerWrapper from './ColorPickerWrapper';

const Harness = () => {
  const handleChange = () => {};

  return (
    <div>
      <div className="closeme"></div>
      <ColorPickerWrapper color={'#fff'} onChange={handleChange} />
    </div>
  );
};

describe('ColorPickerWrapper', () => {
  test('should render correctly', () => {
    const testfn = jest.fn();
    const { container } = render(
      <ColorPickerWrapper color={'#fff'} onChange={testfn} />
    );
    expect(container).toMatchSnapshot();
  });
  test('should handle click', () => {
    const testfn = jest.fn();
    const { container } = render(
      <ColorPickerWrapper color={'#fff'} onChange={testfn} />
    );

    const button = container.querySelector('button') as HTMLButtonElement;
    fireEvent.click(button);
  });
  test('should handle close', () => {
    const { container } = render(<Harness />);

    const button = container.querySelector('button') as HTMLButtonElement;
    fireEvent.click(button);

    const close = container.querySelector('.closeme') as HTMLDivElement;
    fireEvent.click(close);
  });
});
