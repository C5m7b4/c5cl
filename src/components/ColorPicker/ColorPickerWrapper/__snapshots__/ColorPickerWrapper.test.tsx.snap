// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerWrapper should render correctly 1`] = `
<div>
  <div>
    <button
      class="mpw-swatch"
      style="height: 20px; width: 120px; border-radius: 10px; border-width: 5px; border-color: #333; border-style: solid; cursor: pointer; background-color: rgb(255, 255, 255);"
    />
  </div>
</div>
`;
